src/lib/utils/eventFetcher.ts

```diff
@@ -57,17 +58,21 @@ export class EventFetcher {
     TContract extends Contract,
     TEventFilter extends TypedEventFilter<TypedEvent>
   >(
-    addr: string,
```

Comment:
Maybe this should go in the filter objects. It should probably be an optional param - that will break backwards compatiblity though